dali/kernels/test/kernel_test_utils.h

```diff
@@ -29,21 +29,109 @@ struct SimpleKernelTestBase {
 
   template <int i>
   using Input =  std::remove_reference_t<
-      std::tuple_element_t<i, kernels::kernel_inputs<Kernel>>>;
+          std::tuple_element_t<i, kernels::kernel_inputs<Kernel>>>;
 
   template <int i>
-  using Output = std::remove_reference_t<
-      std::tuple_element_t<i, kernels::kernel_outputs<Kernel>>>;
+  using Output = std::remove_reference_t<std::tuple_element_t<i,
+          kernels::kernel_outputs<Kernel>>>;
 
   template <int i>
   using Arg = std::tuple_element_t<i, kernels::kernel_args<Kernel>>;
 
   template <int i>
   using InputElement = std::remove_const_t<element_t<Input<i>>>;
+
   template <int i>
   using OutputElement = element_t<Output<i>>;
 };
 
+namespace detail {
+
+template <template <typename A, typename B> class Pair, typename T1, typename... T>
+using FixedFirstTypePairs = std::tuple<Pair<T1, T>...>;
+
+template <template <typename A, typename B> class Pair, typename A, typename B>
+struct AllPairsHelper;
+
+template <template <typename A, typename B> class Pair, typename... A, typename... B>
+struct AllPairsHelper<Pair, std::tuple<A...>, std::tuple<B...>> {
+  using type = dali::detail::tuple_cat_t<FixedFirstTypePairs<Pair, A, B...>...>;
+};
+
+template <template <typename A, typename B> class Pair, typename TupleA, typename TupleB>
+using AllPairs = typename AllPairsHelper<Pair, TupleA, TupleB>::type;
+
+template <typename Tuple>
+struct TupleToGTest;
+
+template <typename... T>
+struct TupleToGTest<std::tuple<T...>> {
+  using type = ::testing::Types<T...>;
+};
+
+template <class InputType, class OutputType>
+struct InputOutputTypes {
+  using in = InputType;
+  using out = OutputType;
```

Comment:
As above

Reply:
After discussion with @jantonguirao , we've set up `using InputType = In;` variant